### PR TITLE
FHAC-1443: Add more Junit test 

### PR DESCRIPTION
**Pull request by balmeeki**
Added test clas forAdhocQueryTransformHelper
@sjarral112 (optional): feel free to review

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyRequestType;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.util.format.PatientIdFormatUtil;
+import java.util.List;
+import oasis.names.tc.ebxml_regrep.xsd.query._3.AdhocQueryRequest;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.xacml._2_0.context.schema.os.RequestType;
+import oasis.names.tc.xacml._2_0.context.schema.os.ResourceType;
+import oasis.names.tc.xacml._2_0.context.schema.os.SubjectType;
+import org.slf4j.Logger;
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java]: Unused Imports

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package gov.hhs.fha.nhinc.transform.policy;
+
+import gov.hhs.fha.nhinc.common.eventcommon.AdhocQueryRequestEventType;
+
+import gov.hhs.fha.nhinc.common.eventcommon.AdhocQueryResultEventType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyRequestType;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.util.format.PatientIdFormatUtil;
+import java.util.List;
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java]: Unused Import

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package gov.hhs.fha.nhinc.transform.policy;
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:1]: Connect License missing

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+     //   eventA.getDirection()= "inbound";
+        eventA.setDirection("inbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNotNull(result);
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:33]: Please check if the result object is valid or not for inbound.

**Review by balmeeki — COMMENTED** [review]
Added test cases and removed unused imports

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package gov.hhs.fha.nhinc.transform.policy;
+
+import gov.hhs.fha.nhinc.common.eventcommon.AdhocQueryRequestEventType;
+
+import gov.hhs.fha.nhinc.common.eventcommon.AdhocQueryResultEventType;
+//import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:6]: Remove comment import if we don't use it

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Before
+    public void setUp() {
+    }
+    
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+     //   eventA.getDirection()= "inbound";
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:30]: Remove if we don't use them

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        eventA.setDirection("inbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNotNull(result);
+    }
+    
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_outbound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+     //   eventA.getDirection()= "inbound";
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:40]: similar as above

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        eventA.setDirection("outbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNotNull(result);
+    }
+
+    @Test
+    public void testTransformAdhocQueryResponseToCheckPolicy_OutBound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+    	AdhocQueryResultEventType eventB = new AdhocQueryResultEventType();
+     //   eventA.getDirection()= "inbound";
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:50]: Remove if we don't use it

**Review by mhpnguyen — CHANGES_REQUESTED** [review]
In additional to that, please make sure to follow https://connectopensource.atlassian.net/wiki/display/CONNECTWIKI/Coding+templates+and+formatters

File: Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_outbound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+     //   eventA.getDirection()= "inbound";
+        eventA.setDirection("outbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNotNull(result);
+    }
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java:44]: assert for valid outbound object

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

**Comment by TabassumJafri** [comment]
Closing the pull request as another PR is up for the same ticket --> https://github.com/CONNECT-Solution/CONNECT/pull/1395

**Pull request closed without merging**